src/core/processing/qgsprocessingutils.cpp

```diff
@@ -107,6 +108,29 @@ QList<QgsMeshLayer *> QgsProcessingUtils::compatibleMeshLayers( QgsProject *proj
   return layers;
 }
 
+QList<QgsPluginLayer *> QgsProcessingUtils::compatiblePluginLayers( QgsProject *project, bool sort )
```

Comment:
This is calling out for a template based method to condense compatibleRasterLayers/compatibleMeshLayers/compatiblePluginLayers into one unified function

Reply:
Added templated private `compatibleMapLayers` and then use it inside compatibleRasterLayers/compatibleMeshLayers/compatiblePluginLayers.

Reply:
looks great!